gateway/handlers/notifiers.go

```diff
@@ -46,19 +46,26 @@ type PrometheusFunctionNotifier struct {
 }
 
 // Notify records metrics in Prometheus
-func (p PrometheusFunctionNotifier) Notify(method string, URL string, originalURL string, statusCode int, duration time.Duration) {
-	seconds := duration.Seconds()
-	serviceName := getServiceName(originalURL)
+func (p PrometheusFunctionNotifier) Notify(method string, URL string, originalURL string, statusCode int, event string, duration time.Duration) {
+	if event == "completed" {
 
-	p.Metrics.GatewayFunctionsHistogram.
-		WithLabelValues(serviceName).
-		Observe(seconds)
+		seconds := duration.Seconds()
+		serviceName := getServiceName(originalURL)
 
-	code := strconv.Itoa(statusCode)
+		p.Metrics.GatewayFunctionsHistogram.
+			WithLabelValues(serviceName).
+			Observe(seconds)
+
+		code := strconv.Itoa(statusCode)
+
+		p.Metrics.GatewayFunctionInvocation.
+			With(prometheus.Labels{"function_name": serviceName, "code": code}).
+			Inc()
+	} else if event == "started" {
+		serviceName := getServiceName(originalURL)
+		p.Metrics.StartedCounter.WithLabelValues(serviceName).Inc()
```

Comment:
I think it would be nice if the metric name was symmetric with the metric that capture the end of the request `GatewayFunctionInvocation` and  `GatewayFunctionInvocationStarted`  or using `FunctionInvocationCompleted` and `FunctionInvocationStarted` to be even more precise

Reply:
We already capture the end of the request using the existing metric.

What do you suggest? Renaming would be a breaking change, recording double the information would be hideous.

Reply:
I am specifically talking about the variable name, `Metrics.StartedCounter` isn't very descriptive.  It is entirely possible we will want other started counters later

Reply:
Give me a suggestion

Reply:
`gateway_function_invocation_started`?

Reply:
I mean the name of the Go variable and my suggestion was/is `GatewayFunctionInvocationStarted` or `FunctionInvocationStarted`,  so that this code would look like 

```go
p.Metrics.FunctionInvocationStarted.WithLabelValues(serviceName).Inc()
```

Reply:
I've updated as per that suggestion.

Reply:
The diff still looks like this

![image](https://user-images.githubusercontent.com/891889/76515655-3bb87500-645a-11ea-94d8-ad3ffa9fb27d.png)

Reply:
Ok I see what you mean now. I haven't changed that internal variable, I'll do it next.

Reply:
Updated and retrofitted into the other counter